gdal_subsetter/transform.py

```diff
@@ -778,177 +784,161 @@ def stack_multi_file_with_metadata(self, infilelist, outfile):
-        dstfile = "%s/result.tif" % (dstdir)
+        dstfile = os.path.join(dstdir, 'result.tif')
 
         if not os.path.exists(dstfile):
             self.cmd2('mv', srcfile, dstfile)
 
         return dstfile
 
     def reformat(self, srcfile, dstdir):
-        gdal_subsetter_version = "gdal_subsetter_version={gdal_subsetter_ver}".format(
-            gdal_subsetter_ver=self.get_version())
+        gdal_subsetter_version = 'gdal_subsetter_version={self.get_version()}'
         output_mime = self.message.format.process('mime')
-        if not output_mime == "image/png" or output_mime == "image/jpeg":
+        if not output_mime == 'image/png' or output_mime == 'image/jpeg':
             command = ['gdal_edit.py',  '-mo', gdal_subsetter_version, srcfile]
             self.cmd(*command)
         if output_mime not in mime_to_gdal:
-            raise Exception('Unrecognized output format: ' + output_mime)
-        if output_mime == "image/tiff":
+            raise Exception(f'Unrecognized output format: {output_mime}')
+        if output_mime == 'image/tiff':
             return srcfile
-        dstfile = "%s/translated.%s" % (dstdir, mime_to_extension[output_mime])
-        if output_mime == "application/x-netcdf4" or output_mime == "application/x-zarr":
-            dstfile = "%s/translated.%s" % (dstdir, "nc")
+
+        dstfile = os.path.join(
+            dstdir, f'translated.{mime_to_extension[output_mime]}'
+        )
+
+        if output_mime in ('application/x-netcdf4', 'application/x-zarr'):
+            dstfile = os.path.join(dstdir, 'translated.nc')
             dstfile = self.geotiff2netcdf_direct(srcfile, dstfile)
             return dstfile
-        else:  # png, jpeg, gif, etc.
-            if os.path.exists(os.path.splitext(srcfile)[0] + '.wld'):
+        else:
+            # png, jpeg, gif, etc.
+            if os.path.exists(f'{os.path.splitext(srcfile)[0]}.wld'):
                 # don't need to reformat files that have been processed earlier
                 return srcfile
-            command = ['gdal_translate',
-                       '-of', mime_to_gdal[output_mime],
-                       '-scale',
-                       srcfile, dstfile]
+
+            command = ['gdal_translate', '-of', mime_to_gdal[output_mime],
+                       '-scale', srcfile, dstfile]
             self.cmd(*command)
             return dstfile
 
 
     def read_layer_format(self, collection, filename, layer_id):
-        gdalinfo_lines = self.cmd("gdalinfo", filename)
-
-        layer_line = next(
-            filter(
-                (lambda line: re.search(":*" + layer_id+"$", line)),
-                gdalinfo_lines
-            ),
-            None
-        )
+        gdalinfo_lines = self.cmd('gdalinfo', filename)
+
+        layer_line = next((line for line in gdalinfo_lines
+                           if re.search(f'SUBDATASET.*:*{layer_id}$', line)
+                           is not None), None)
```

Comment:
This is the second of the two more significant changes I made this time around. I noticed that the old version of the HGA regression Sentinel test-case 13 was failing for a silly reason. There is a variable (that probably shouldn't be processed anyway) in the root group called `productBoundingBox`. In the `gdalinfo` output the first line that referred to that variable was part of the metadata:

```
NC_GLOBAL#ogr_layer_name=productBoundingBox
```

However, the thing that should have matched was the subsequent line:

```
SUBDATASET_1_NAME=NETCDF:"/Users/olittlej/Downloads/S1-GUNW-A-R-166-tops-20200313_20200206-014119-34455N_32574N-PP-1749-v2_0_2.nc":productBoundingBox
```

This change guards against that. (It also drops the lambda function in favour of a generator, which I think(?) is considered more "Pythonic")